optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/score/descriptor/ScoreDescriptor.java

```diff
@@ -0,0 +1,238 @@
+                return new HardMediumSoftLongScoreDefinition();
+            } else if (scoreType.equals(HardMediumSoftBigDecimalScore.class)) {
+                return new HardMediumSoftBigDecimalScoreDefinition();
+            } else {
+                throw new IllegalArgumentException("The solutionClass (" + solutionClass
+                        + ") has a @" + PlanningScore.class.getSimpleName()
+                        + " annotated member (" + scoreMemberAccessor
+                        + ") that returns a scoreType (" + scoreType
+                        + ") that is not recognized as a default " + Score.class.getSimpleName() + " implementation.\n"
+                        + "  If you intend to use a custom implementation,"
+                        + " maybe set a scoreDefinition in the @" + PlanningScore.class.getSimpleName()
+                        + " annotation.");
+            }
+        } else {
+            int bendableHardLevelsSize = annotation.bendableHardLevelsSize();
+            int bendableSoftLevelsSize = annotation.bendableSoftLevelsSize();
+            if (bendableHardLevelsSize == PlanningScore.NO_LEVEL_SIZE
+                    || bendableSoftLevelsSize == PlanningScore.NO_LEVEL_SIZE) {
+                throw new IllegalArgumentException("The solutionClass (" + solutionClass
+                        + ") has a @" + PlanningScore.class.getSimpleName()
+                        + " annotated member (" + scoreMemberAccessor
+                        + ") that returns a scoreType (" + scoreType
+                        + ") that must have a bendableHardLevelsSize (" + annotation.bendableHardLevelsSize()
+                        + ") and a bendableSoftLevelsSize (" + annotation.bendableSoftLevelsSize() + ").");
+            }
+            if (scoreType.equals(BendableScore.class)) {
+                return new BendableScoreDefinition(bendableHardLevelsSize, bendableSoftLevelsSize);
+            } else if (scoreType.equals(BendableLongScore.class)) {
+                return new BendableLongScoreDefinition(bendableHardLevelsSize, bendableSoftLevelsSize);
+            } else if (scoreType.equals(BendableBigDecimalScore.class)) {
+                return new BendableBigDecimalScoreDefinition(bendableHardLevelsSize, bendableSoftLevelsSize);
+            } else {
+                throw new IllegalArgumentException("The solutionClass (" + solutionClass
+                        + ") has a @" + PlanningScore.class.getSimpleName()
+                        + " annotated member (" + scoreMemberAccessor
+                        + ") that returns a bendable scoreType (" + scoreType
+                        + ") that is not recognized as a default " + Score.class.getSimpleName() + " implementation.\n"
+                        + "  If you intend to use a custom implementation,"
+                        + " maybe set a scoreDefinition in the annotation.");
+            }
```

Comment:
Personally, I believe this belongs on `ScoreDefinition`. Something like `public static Optional<ScoreDefinition> findFor(Class<Score> scoreClass)`.

The exception thrown can then be customized at the call site which receives the empty `Optional`.

Reply:
The idea is nice but `ScoreDefinition` construction is not that simple because 1) bendable score definitions require `hardLevelSize` and `softLevelSize`, 2) the exception messages require additional information that is not related to score defintion (solution class, score member).